### PR TITLE
scylla_swap_setup: run error check before allocating swap and increase swap allocation speed

### PR DESCRIPTION
**Pull request by syuu1228**
This patch fixes error check and speed up swap allocation.

Following patches are included:
 - scylla_swap_setup: run error check before allocating swap
   avoid create swapfile before running error check
 - scylla_swap_setup: use fallocate on ext4
   this inclease swap allocation speed on ext4

### CODE REVIEW

**Comment by syuu1228** [comment]
I changed my mind, `--overwrite-unit-file` is too Azure (or cloud) specific feature, not related with on-premise users.
I think it's better to handle in scylla-machine-image, not here.
I will rebase this patchset without `scylla_swap_setup: allow overwrite systemd unit file`.

**Comment by syuu1228** [comment]
Rebased, dropped `scylla_swap_setup: allow overwrite systemd unit file`.

**Comment by syuu1228** [comment]
ping

**Comment by syuu1228** [comment]
@avikivity ping

**Comment by syuu1228** [comment]
@avikivity ping, we want this for Azure machine-image

File: dist/common/scripts/scylla_swap_setup [hunk 1]
```diff
@@ -42,6 +55,18 @@ if __name__ == '__main__':
...
 
+    swap_directory = Path(args.swap_directory)
+    swapfile =  swap_directory / 'swapfile'
+    if swapfile.exists():
+        print('swapfile {} already exists'.format(swapfile))
+        sys.exit(1)
+
+    swapunit_bn = out('systemd-escape -p --suffix=swap {}'.format(swapfile))
+    swapunit = Path('/etc/systemd/system/{}'.format(swapunit_bn))
+    if not args.overwrite_unit_file and swapunit.exists():
```
Comment (benipeled) [inline dist/common/scripts/scylla_swap_setup:66]: There is no `overwrite_unit_file` arg

**Comment by benipeled** [comment]
@avikivity please dequeue (It fails scylla installation - https://github.com/scylladb/scylladb/issues/16331)

**Pull request merged**